src/pydata_sphinx_theme/__init__.py

```diff
@@ -264,6 +284,13 @@ def _add_collapse_checkboxes(soup):
         # We check all "li" elements, to add a "current-page" to the correct li.
         classes = element.get("class", [])
 
+        # expanding the parent part explicitly, if present
+        if "current" in classes:
+            parentli = element.find_parent("li", class_="toctree-l0")
+            if parentli:
+                caption = parentli.find("p", class_="caption")
+                caption.find_next_sibling("input").attrs["checked"] = ""
```

Comment:
```suggestion
                caption = parentli.select("p.caption ~ input").attrs["checked"] = ""
```

again we should double-check this syntax, but I think this would work as a more CSS-native one-liner